### PR TITLE
Implemented basic entity and dynamic-routing

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT

#### What's in this PR?

This PR implements very basic the needed entity and dynamic-router.

#### Why?

This is the basic for all the upcoming tasks.

### CODE REVIEW

File: Resources/config/services.xml [hunk 1]
```diff
@@ -10,5 +10,11 @@
             <tag name="sulu.admin"/>
             <tag name="sulu.context" context="admin"/>
         </service>
+
+        <service id="sulu.repository.redirect_route" class="Sulu\Bundle\RedirectBundle\Entity\RedirectRouteRepository">
```
Comment (wachterjohannes) [inline Resources/config/services.xml]: replace with sulu persistence

File: Controller/RedirectController.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     *
+     * @param Request $request
+     * @param RedirectRouteInterface $redirectRoute
+     * @param $resourceLocatorPrefix
+     *
+     * @return RedirectResponse
+     */
+    public function redirect(Request $request, RedirectRouteInterface $redirectRoute, $resourceLocatorPrefix)
+    {
+        // TODO query parameter
```
Comment (wachterjohannes) [inline Controller/RedirectController.php]: add query parameter to resulting route

File: Resources/config/doctrine/RedirectRoute.orm.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
+                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
+    <mapped-superclass name="Sulu\Bundle\RedirectBundle\Entity\RedirectRoute"
+            repository-class="Sulu\Bundle\RedirectBundle\Entity\RedirectRouteRepository" table="re_routes">
+        <indexes>
+            <index columns="source"/>
```
Comment (alexander-schranz) [inline Resources/config/doctrine/RedirectRoute.orm.xml]: is not needed when its a unique constraints

File: Resources/config/routing.xml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                 class="Symfony\Cmf\Component\Routing\NestedMatcher\NestedMatcher">
+            <argument type="service" id="sulu_redirect.routing.provider"/>
+            <argument type="service" id="sulu_redirect.routing.final_matcher"/>
+        </service>
+
+        <service id="sulu_redirect.routing.router" class="Symfony\Cmf\Bundle\RoutingBundle\Routing\DynamicRouter">
+            <argument type="service" id="router.request_context"/>
+            <argument type="service" id="sulu_redirect.routing.nested_matcher"/>
+            <argument type="service" id="sulu_redirect.routing.generator"/>
+            <argument>%sulu_redirect.uri_filter_regexp%</argument>
```
Comment (alexander-schranz) [inline Resources/config/routing.xml]: set it directly here to null

File: Controller/RedirectController.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+     * @param $resourceLocatorPrefix
+     *
+     * @return RedirectResponse
+     */
+    public function redirect(Request $request, RedirectRouteInterface $redirectRoute, $resourceLocatorPrefix = '')
+    {
+        $queryString = http_build_query($request->query->all());
+        $url = [
+            $request->getSchemeAndHttpHost(),
+            $resourceLocatorPrefix,
```
Comment (chirimoya) [inline Controller/RedirectController.php]: ignore webspaces & languages for now

**Pull request merged**